gateway/buckets.go

```diff
@@ -84,9 +83,12 @@ func (g *Gateway) renderBucketPath(c *gin.Context, ctx context.Context, threadID
 		return
 	}
 	if !rep.Item.IsDir {
-		if err := g.buckets.PullPath(ctx, buck.Key, pth, c.Writer); err != nil {
+		contentType, r, err := getContentTypeFromPullPath(ctx, g.buckets, buck.Key, pth)
+		if err != nil {
 			render404(c)
 		}
+		c.Writer.Header().Set("Content-Type", contentType)
+		io.Copy(c.Writer, r)
```

Comment:
The main idea is always relying on `getContentTypeFromPullPath` which returns the detected content type, and an `io.Reader` to copy to the response writer.